ceno_zkvm/src/e2e.rs

```diff
@@ -159,33 +160,79 @@ fn emulate_program(
     }
 }
 
-fn init_mem(
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+pub enum Preset {
+    Ceno,
+    Sp1,
+}
+
+pub fn setup_platform(
+    preset: Preset,
     program: &Program,
-    platform: &Platform,
-    mem_padder: &mut MemPadder,
     stack_size: u32,
     heap_size: u32,
-) -> Vec<MemInitRecord> {
-    let stack_addrs = platform.stack_top - stack_size..platform.stack_top;
-    // Detect heap as starting after program data.
-    let heap_start = program.image.keys().max().unwrap() + WORD_SIZE as u32;
-    let heap_addrs = heap_start..heap_start + heap_size;
+    pub_io_size: u32,
+) -> Platform {
+    let preset = match preset {
+        Preset::Ceno => CENO_PLATFORM,
+        Preset::Sp1 => Platform {
+            // The stack section is not mentioned in ELF headers, so we repeat the constant STACK_TOP here.
+            stack: 0x0020_0400..0x0020_0400,
+            unsafe_ecall_nop: true,
+            ..CENO_PLATFORM
+        },
+    };
+
+    let prog_data = program.image.keys().copied().collect::<HashSet<Addr>>();
+    let stack = preset.stack.end - stack_size..preset.stack.end;
+    let heap = {
+        // Detect heap as starting after program data.
+        let heap_start = program.image.keys().max().unwrap() + WORD_SIZE as u32;
+        let heap = heap_start..heap_start + heap_size;
+        // Pad the total size to the next power of two.
```

Comment:
Please compare the discussion around https://github.com/scroll-tech/ceno/pull/716

Let's pick one policy for whether we should round up or down or assert, and decide one ideally one place where we do that.

Reply:
How do you know it's not related, if you don't understand it?